ZEP000-wf-stream-processing.md

```diff
@@ -0,0 +1,858 @@
+* since the command `StartEvent:Element_Completing` produces the event `Sequence_Flow_Taken`, it must produce the next command `ServiceTask:Element_Activating` to activate the target activity of the taken sequence flow
+
+On processing:
+* the `WorkflowInstanceCreationIntent:Create` command creates the new workflow
+  instance
+* the `Process:Element_Activating` command transitions the process element to
+  activated
+* the `StartEvent:Element_Activating` command transitions the start event to
+  activated
+* the `StartEvent:Element_Completing` command transitions the start event to
+  completed and marks the sequence flow as taken
+
+On replay:
+* the `WorkflowInstanceCreationIntent:Created` event creates the new workflow
+  instance
+* the `Process:Element_Activated` event transitions the process element to
+  activated
+* the `StartEvent:Element_Activated` event transitions the start event to
+  activated
+* the `StartEvent:Element_Completed` event transitions the start event to
+  completed
+* the `Sequence_Flow_Taken` event marks the sequence flow as taken
+* processing can continue again at `ServiceTask:Element_Activating`
+
+Differences to current processing:
+* BPMN elements can no longer be in the `activating` or `completing` state.
+  Activating and completing have moved to commands that when processed result in
+  `activated` and `completed` states both in state as well as on the log stream
+* the `...:Element_Activated`, `...:Element_Completed` and `Sequence_Flow_Taken`
+  events no longer produce follow-up records, because they are now strictly
+  events
+
+### Service Task
+
+Command to Process | Written Events and Follow-up Commands
+---|---
+ServiceTask:Element_Activating | ServiceTask:Element_Activated, [Variable:Created], Job:Created
+JobBatch:Activate | JobBatch:Activated 
+Job:Complete | Job:Completed, ServiceTask:Element_Completing
+ServiceTask:Element_Completing | [Variable:Created], [Variable:Updated], ServiceTask:Element_Completed, Sequence_Flow_Taken, EndEvent:Element_Activating,
+
+Notes:
+* if the service task has an input variable mapping then it produces one or more `Variable:Created` events for the new local variables
+* the event `Job:Created` is produced by the command `ServiceTask:Element_Activating` instead of a `Job:Create` command because it can't be interrupted
+* if the job is completed with variables then these variables will be stored in the state until they are used for the output mapping (i.e. temporal/transient variables). They must therefore also be written into the `Job:Completed` event.
+* if the service task has an output variable mapping, or the job is completed with variables, then it produces one or more `Variable:Created`/`Variable:Updated` events for the new and updated variables of the workflow instance or scope
+* the order of writing follow-up events matters. Both `ServiceTask:Element_Activated` and `Variable:Created` need to be replayed before `Job:Created`.
```

Comment:
It would also work in this case. In general, there is logical order of things, for example, apply input mappings before creating the job. In other cases, there is a dependency between the records, for example, `Element_Activating` before `Variable_Create`. 

Let's describe the ordering shortly in the guidance section.